private/buf/bufsync/bufsync.go

```diff
@@ -175,13 +175,13 @@ type Handler interface {
 		hash git.Hash,
 	) (bool, error)
 
-	// GetModuleReleaseBranch is invoked before syncing, to gather release branch names for all the
-	// modules that are about to be synced. If the BSR module does not exist, the implementation should
-	// return `ModuleDoesNotExistErr` error.
-	GetModuleReleaseBranch(
+	// IsProtectedBranch is invoked when syncing branches to know if a branch's history is protected
+	// and must not diverge since the last sync. If an error is returned, sync will abort.
+	IsProtectedBranch(
```

Comment:
Nit: we could invoke just once per module identity and get `protectedBranches []string`

Reply:
We could, but the flow isn't like that right now. We cache anyway, so this is fine.